fission/src/mirabuf/MirabufSceneObject.ts

```diff
@@ -459,6 +464,18 @@ class MirabufSceneObject extends SceneObject implements ContextSupplier {
         }
     }
 
+    public RemoveScoringZoneObject(zone: ScoringZonePreferences) {
+        const index = this._fieldPreferences?.scoringZones?.indexOf(zone) ?? -1
+        if (index != -1) {
+            const zoneObject = this._scoringZones[index]
+
+            if (zoneObject != null) {
+                World.SceneRenderer.RemoveSceneObject(zoneObject.id)
+                zoneObject.id = -1
+            }
+        }
```

Comment:
In my opinion, unnecessary nesting is an anti-pattern.

https://youtu.be/CFRhGnuXG-4

Reply:
I tend to agree with this, for things this small I do tend think its up to the discretion of the author to determine what looks the best. Not going to have a hard rule on this except for more extreme cases.